lib/Models/ArcGisMapServerCatalogItem.ts

```diff
@@ -143,43 +151,26 @@ class MapServerStratum extends LoadableStratum(
     }
 
     // TODO: if tokenUrl, fetch and pass token as parameter
-    const serviceMetadata = getJson(item, serviceUri);
-    const layersMetadata = getJson(item, layersUri);
-    const legendMetadata = getJson(item, legendUri);
+    const serviceMetadata = await getJson(item, serviceUri);
+    let layersMetadata = await getJson(item, layersUri);
+    const legendMetadata = await getJson(item, legendUri);
 
-    const results = await Promise.all([
-      serviceMetadata,
-      layersMetadata,
-      legendMetadata
-    ]);
-
-    const mapServer = results[0];
-    const legend = results[2];
-
-    let allLayers;
-    if (isDefined(results[1].layers)) {
-      allLayers = results[1].layers;
-    } else if (isDefined(results[1].id)) {
-      // single layer
-      allLayers = [results[1]];
+    // Use the slightly more basic layer metadata
+    if (layersMetadata === undefined) {
+      layersMetadata = serviceMetadata.layers;
     } else {
-      throw new TerriaError({
-        title: i18next.t(
-          "models.arcGisMapServerCatalogItem.unusableMetadataTitle"
-        ),
-        message: isDefined(results[0].error)
-          ? results[0].error.message
-          : i18next.t(
-              "models.arcGisMapServerCatalogItem.unusableMetadataDefaultMessage"
-            )
-      });
+      if (layersMetadata.layers !== undefined) {
+        layersMetadata = layersMetadata.layers;
+      } else if (layersMetadata.id) {
+        layersMetadata = [layersMetadata];
+      }
     }
 
     const stratum = new MapServerStratum(
       item,
-      mapServer,
-      allLayers,
-      legend,
+      serviceMetadata,
+      layersMetadata,
```

Comment:
`layersMetadata` is now `Layer[] | LayerSummary[] | undefined` but the type signature for the `MapServerStratum` constructor is `Layer[]`, also `serviceMetadata` or `legendMetadata` could be undefined.
Would it be worth making this all a bit more ts friendly? I'd be happy to do it (I know how much you love ts)

Reply:
I also think it would be worth throwing an error if any of those are `undefined` (assuming they are all needed to be defined)

Reply:
`serviceMetadata` is a must, `layersMetadata ` will always be populated although the quality will vary, and `legendMetadata` is a maybe

Reply:
@nf-s yeah the TS side of it is pretty clunky so open to ideas. Perhaps I'll talk you through the scenarios tomorrow so you can get a feel fo why it's so.